src/controllers/authController.js

```diff
@@ -0,0 +1,32 @@
+import passport from 'passport';
+import JWT from 'jsonwebtoken';
+import { config } from 'dotenv';
+import passportConfIgnored from '../config/passport';
+
+config();
+export const callback = (provider) => {
+  const callbackFunction = (req, res, next) => {
+    if (req.headers.testunauthorized) provider = 'unauthorized';
+    passport.authenticate(
+      provider,
+      (err, user) => {
+        if (!user) {
+          return res.status(401).send({ status: 401, message: 'You\'re not authenticated' });
+        }
+        const token = JWT.sign({
+          iss: 'Wolfsbane-BarefootNomad',
+          sub: user,
+        }, process.env.APP_SECRET);
```

Comment:
Token creation and verification should be abstracted to a function in the `utils` that can be imported and used anywhere.

Reply:
I used it here like this because I didn't want to duplicate methods. @oyewoas  is already writing that method, but his work hasn't been merged yet.